runner/src/internal.rs

```diff
@@ -280,7 +353,7 @@ pub async fn run_step(context: &Context, step: Step) -> StepResult {
             let background_stdout_future = tokio::spawn(read_to_end(running_background.stdout()));
             let background_stderr_future = tokio::spawn(read_to_end(running_background.stderr()));
 
-            let mut foreground_result = run_step(&context, *foreground).await;
+            let mut foreground_result = run_step(&context, *foreground, run_status.clone()).await;
```

Comment:
If the `run_status` is changed in the `foreground`, should it influence the `run_status` value (why is it cloned)?
For example if there were failures inside `foreground`.